setup.py

```diff
@@ -63,20 +64,43 @@
 console_scripts = ['aria = aria.cli.cli:main']
 
 
-class InstallCommand(install):
-    user_options = install.user_options + [
+def _generate_user_options(command):
+    return command.user_options + [
         ('skip-ctx', None, 'Install with or without the ctx (Defaults to False)')
     ]
-    boolean_options = install.boolean_options + ['skip-ctx']
 
-    def initialize_options(self):
-        install.initialize_options(self)
-        self.skip_ctx = False
 
-    def run(self):
-        if self.skip_ctx is False:
-            console_scripts.append('ctx = aria.orchestrator.execution_plugin.ctx_proxy.client:main')
-        install.run(self)
+def _generate_boolean_options(command):
+    return command.boolean_options + ['skip-ctx']
+
+
+def _initialize_options(custom_cmd):
+    custom_cmd.command.initialize_options(custom_cmd)
+    custom_cmd.skip_ctx = False
+
+
+def _run(custom_cmd):
+    if custom_cmd.skip_ctx is False:
+        console_scripts.append('ctx = aria.orchestrator.execution_plugin.ctx_proxy.client:main')
+    custom_cmd.command.run(custom_cmd)
+
+
+class InstallCommand(install):
```

Comment:
why not have a base CustomCommand class,
have all the private functions in it as well,
and have two subclasses which only set `command`?